db/database.go

```diff
@@ -162,6 +162,13 @@ func (d *PgxDatabase) Transact(fn TxFunc, opts ...TxOpt) (interface{}, error) {
 	}
 	var attempt int
 	var ret interface{}
+	var tx pgx.Tx
+	defer func() {
+		if p := recover(); p != nil && tx != nil {
+			_ = tx.Rollback(options.ctx)
```

Comment:
Consider reporting this error, e.g.
```suggestion
			err = tx.Rollback(options.ctx)
			if err != nil {
				p = fmt.Errorf("%w, and %s while rolling back", p, err)
			}
```
If something *does* go wrong rolling back during a panic, more information is better.

Reply:
Currently, we do no log rollback in the Transaction function. We can capture the information and the stack trace in the top-level handler if needed.
But what will be the added value of the rollback error as we assume everything is aborted in this flow.